cmd/object-api-utils.go

```diff
@@ -170,6 +170,13 @@ func IsValidObjectPrefix(object string) bool {
 	if strings.Contains(object, `//`) {
 		return false
 	}
+	// This is valid for AWS S3 but it will never
+	// work with file systems, we will reject here
+	// to return object name invalid rather than
+	// a cryptic error from the file system.
+	if strings.Contains(object, "\x00") {
```

Comment:
```suggestion
	if !utf8.Valid(object) {
```
That should be a more generic validation. `import "unicode/utf8"` is probably needed.

Reply:
@klauspost \x00 is utf8 valid, also we are doing that in line 167

Reply:
I thought doing this adds latency to the incoming calls or no? - one of the reasons had avoided using utf8 in this codepath.

Reply:
Really? Wow